### PR TITLE
Consent Enhancements: Privacy, Personalization and Marketing preferences

### PR DESCRIPTION
**Pull request by RussStringham**
Please link to the issue #…

### CODE REVIEW

File: schemas/context/experienceevent-privacy.schema.json [hunk 1]
```diff
@@ -18,6 +18,10 @@
    "definitions":{
       "experienceevent-privacy":{
          "properties":{
+            "xdm:consentsAndPreferneces":{
```
Comment (ankushkd) [inline schemas/context/experienceevent-privacy.schema.json:21]: Key spelling.

File: docs/standards.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ XDM incorporates or builds a number of existing standards and specifications.
 Prior to XDM, Adobe defined and standardized the Extensible Metadata Platform (XMP) throughout its Creative Cloud products.
```
Comment (lrosenthol) [inline docs/standards.md:25]: Not just CC - but DC and others.  That is also why we took XMP to ISO for standardization (ISO 16684-1)

File: docs/standards.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ XDM incorporates or builds a number of existing standards and specifications.
...
 The vocabularies and operations of XMP entities are essential for those products as is XDM for the wider experience technology ecosystem.
 
-In XDM, these two efforts are integrated via the Resource Description Framework (RDF), which is a Word Wide Web Consortium (W3C) standard for data model interchange.
+In XDM, these two efforts are integrated via the Resource Description Framework (RDF), which is a World Wide Web Consortium (W3C) standard for data model interchange.
 XMP models are, at their core, RDF models expressed in an XML-based serialization.
```
Comment (lrosenthol) [inline docs/standards.md:29]: And the fact is that XMP (and RDF) is serialization agnostic.  The new  ISO 16684-3 provides for a JSON-LD-based serialization of the XMP data model.

File: docs/standards.md [hunk 4]
```diff
@@ -25,7 +25,7 @@ XDM incorporates or builds a number of existing standards and specifications.
...
 XDM models are, at their core, also RDF models, expressed in JSON-LD.
 Either one can be translated to the other using the underlying RDF model as an intermediary.
```
Comment (lrosenthol) [inline docs/standards.md:31]: Provided that you follow the rules laid out in ISO 16684-3!

File: schemas/context/profile-privacy.example.1.json [hunk 5]
```diff
@@ -2,11 +2,74 @@
...
+		"xdm:marketingPreferences": {
+			"xdm:defaultMarketingPreference": {
+				"xdm:defaultMarketingChoice": "unknown"
+			},
+			"xdm:marketingOptions": [
+				{
+					"xdm:marketingType": "email",
+					"xdm:marketingChoice": "in",
+                    "xdm:marketingSubscriptions": {
+                       "weekly_mailer": {
```
Comment (lrosenthol) [inline schemas/context/profile-privacy.example.1.json:47]: this isn't a valid XDM key name.  It needs to be namespaced and camelCased.

File: schemas/context/profile-privacy.example.1.json [hunk 6]
```diff
@@ -2,11 +2,74 @@
...
+                          "xdm:subscriptionChoice": "out",
+                          "xdm:subscriptionTimestamp": "2019-02-03T15:52:25+00:00"
+                       },
+                       "daily_newsletter": {
```
Comment (lrosenthol) [inline schemas/context/profile-privacy.example.1.json:51]: this isn't a valid XDM key name.  It needs to be namespaced and camelCased.

File: schemas/context/profile-privacy.example.1.json [hunk 7]
```diff
@@ -2,11 +2,74 @@
...
+                       }
+                    }
+				},
+				{
+					"xdm:marketingType": "iot",
+					"xdm:marketingChoice": "out",
+					"xdm:marketingTimestamp": "2019-01-01T15:52:25+00:00",
+					"xdm:marketingBasisOfProcessing": "legitimate_interest",
+                    "xdm:marketingSubscriptions": {
+                       "out_of_milk": {
```
Comment (lrosenthol) [inline schemas/context/profile-privacy.example.1.json:62]: this isn't a valid XDM key name.  It needs to be namespaced and camelCased.

File: schemas/context/consentpreferences.example.1.json [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+           "xdm:basisOfProcessing": "legitimate_interest",
+           "xdm:subscriptions": {
+              "out_of_milk": {
+                 "xdm:choice": "in"
+              }
+           }
+        }
+     ]
+  },
+  "xdm:version": "1.0.0",
```
Comment (khanna-vikas) [inline schemas/context/consentpreferences.example.1.json:69]: These 4 data elements are under consent preferences?

File: schemas/context/profile-privacy.example.1.json (hunk 5 above)
Comment (cmathis) [reply schemas/context/profile-privacy.example.1.json:47]: This "xdm:subscriptions" is defined as a "Map" datatype, so this is example is showing a potential key in that map. In XDM (and they way AEP uses Maps) the key is simply a "string". The "value" portion of that Map field has a schema in which case those fields are namespaced and defined as needed.

**Comment by prabhum2** [comment]
@RussStringham  - Can you please resolve above conflicts

File: schemas/context/profile-privacy.schema.json [hunk 9]
```diff
@@ -1,89 +1,46 @@
...
             "meta:xdmType": "map",
             "additionalProperties": {
               "type": "object",
               "title": "User Identity",
               "properties": {
+                "xdm:consentsAndPreferences":{
+                  "title": "Device/Identity-specific Consents, Personalization and Marketing Preferences",
+                  "ref": "https://ns.adobe.com/xdm/context/consentpreferences"
+                },
                 "xdm:identityIABConsent": {
```
Comment (ankushkd) [inline schemas/context/profile-privacy.schema.json:44]: here xdm:identityIABConsent and xdm:consentsAndPreferences are parallel but in associated example xdm:identityIABConsent seems to be child of xdm:consentsAndPreferences.

**Pull request merged**